toffy/streak_detection_test.py

```diff
@@ -0,0 +1,178 @@
+                    streak_data=fov.streak_data,
+                    corrected_channels=fov.corrected_channels,
+                    data_dir=save_dir,
+                    save_streak_data=True,
+                )
+            # Gather all the saved files
+            fovs = [fov.stem for fov in save_dir_path.glob("*//") if "corrected" in fov.stem]
+
+            # Check that the names in the corrected files are the same as the original files
+
+            for corrected_fov, fov in zip(fovs, self.fovs):
+                corrected_tiff_names = [
+                    cor_tiff.stem
+                    for cor_tiff in Path(save_dir_path / corrected_fov).glob("*.tiff")
+                ]
+
+                # Get initial file names
+                initial_tiff_names = [
+                    init_tiff.stem
+                    for init_tiff in Path(self.data_dir / fov.fov_name).glob("*.tiff")
+                ]
+
+                # Test that the set of initial tiffs map 1-to-1 to set of corrected tiffs
+                assert Counter(initial_tiff_names) == Counter(corrected_tiff_names)
+
+
+# ? Construct different sets of fake data (noisy, and clean variants)
+# ? Modify to work with `parameterize_with_cases`
+# TODO: All channels of fake data must contain the same streaks. Add noise to each channel.
+# TODO: Add multiple streaks.
+# TODO: Hardcode the streak tests first, request for review
+@pytest.fixture(scope="class", params=[2, 4, 10])
+def fake_data(request):
+    # Set up fake data dimensions
+    num_images = 10
+    row_size = 1000
+    col_size = 1000
+    fake_data = np.zeros(shape=(row_size, col_size, num_images), dtype=np.uint8)
+
+    # Initialize a new generator
+    rng = np.random.default_rng(12345)
+
+    # Generate Streaks
+    for _ in range(request.param):
+        (x_min,) = rng.integers(low=0, high=col_size - 50, size=1)
+        (x_max,) = rng.integers(low=x_min + 50, high=col_size - 1, size=1)
+        (y,) = rng.integers(low=0, high=row_size - 1, size=1)
+        fake_streak = np.ones(shape=(x_max - x_min, num_images))
+        fake_data[y, x_min:x_max, :] = fake_streak
+
+    # for fd in range(num_images):
+
+    # Create fake DataArray
+    fake_data_da = xr.DataArray(
+        data=fake_data,
+        coords=[range(row_size), range(col_size), range(num_images)],
+        dims=["rows", "cols", "channels"],
+    )
+    # Create fake StreakData dataclass
+    # Use channel 0 as the streak detection channel.
+    streak_data: sd.StreakData = sd.StreakData(
+        shape=(row_size, col_size), fov="fake_fov", streak_channel=0
+    )
+
+    # Add the Fake Fov to the request object
+    request.cls.fake_fov = Fov(
+        fov_name="fake_fov", fov_da=fake_data_da, corrected_channels=None, streak_data=streak_data
+    )
+
+
+@pytest.mark.usefixtures("fake_data")
+class TestHelperFunctions:
+    def test_make_binary_mask(self):
+
+        for channel in self.fake_fov.fov_da.channels.values:
```

Comment:
The `_make_binary_mask` function takes a single channel as an input, and returns a single mask as an output. Therefore, when we're testing this function, we should have the minimum amount of infrastructure needed to support it. Instead of creating a class which reads in FOVs, and then looping over each channel in in the  FOV, we should just provide a single channel and check the output.